kythe/go/util/riegeli/compression.go

```diff
@@ -56,8 +56,13 @@ type byteReadCloser struct{ io.ReadCloser }
 // ReadByte implements the io.ByteReader interface.
 func (b byteReadCloser) ReadByte() (byte, error) {
 	var buf [1]byte
-	_, err := b.Read(buf[:])
-	return buf[0], err
+	for {
```

Comment:
This seems dangerous. What circumstances lead to the need for a retry? Can that happen more than once?

I'm concerned that if we hit the pathological state, this could turn into a CPU spinner. I wonder if we could get the same benefit from just "trying again" and failing if it fails the second time.

Reply:
I converted to using `io.ReadFull`.  It does the exact same thing, but it looks less awful.